### PR TITLE
integration-tests: extend find command tests

### PR DESCRIPTION
**Pull request by fgimenez**
Added tests according to the recommendations from the Ubuntu Personal - Product Test team. A reference to the Test Case ID is included in each test comment.

### CODE REVIEW

File: integration-tests/tests/snap_find_test.go [hunk 1]
```diff
@@ -47,3 +63,50 @@ func (s *searchSuite) TestSearchMustPrintMatch(c *check.C) {
...
+	actual := cli.ExecCommand(c, "snap", "find", "--help")
+
+	c.Assert(actual, check.Matches, expected)
+}
+
+// SNAP_FIND_007: find packages with search string containing special characters
+func (s *searchSuite) TestFindWithSpecialCharsInSearchString(c *check.C) {
+	specialChars := "!@#$%^&*/=[]+_:;,.?{}"
+
+	for _, char := range specialChars {
```
Comment (mvo5) [inline integration-tests/tests/snap_find_test.go]: This is a bit surprising. So searching for `_` finds all snaps in the store? That smeels a bit like a store bug/misbehaviour, I'm not sure we want to have a test for this. Or am I misreading the test or the intention behind it :) ?

**Comment by mvo5** [comment]
Thanks for this branch, its super nice that we have a test for `snap find hello-world` now, i.e. to ensure that the store does not regress on `-` again! One question about the last test (SNAP_FIND_007) otherwise :+1:

File: integration-tests/tests/snap_find_test.go [hunk 2]
```diff
@@ -47,3 +63,50 @@ func (s *searchSuite) TestSearchMustPrintMatch(c *check.C) {
 		c.Check(searchOutput, check.Matches, expected)
 	}
 }
+
+// SNAP_FIND_001: list all packages available on the store
```
Comment (mvo5) [inline integration-tests/tests/snap_find_test.go]: Just curious about the prefix with SNAP_FIND_001 - is that referencing some test case that was written up somewhere else?

File: integration-tests/tests/snap_find_test.go (hunk 1 above)
Comment (fgimenez) [inline integration-tests/tests/snap_find_test.go]: You are right, the test reflects the current behaviour, `snap find _` returns the complete list of snaps. Should report a bug about this? (if yes, where?) If this is not the intended way of working I could also reverse the test (for the special characters do not show any results) and disable it until the bug if fixed.

Thanks!

Comment (mvo5) [inline integration-tests/tests/snap_find_test.go]: I reported https://bugs.launchpad.net/click-package-index/+bug/1583952 and I think its best to disable/skip the test for now with a reference to the bug in question. Lets see what the store people have to say about this :)

Comment (fgimenez) [inline integration-tests/tests/snap_find_test.go]: Done, thanks! :)

File: integration-tests/tests/snap_find_test.go (hunk 2 above)
Comment (niemeyer) [inline integration-tests/tests/snap_find_test.go]: Good question. Was also wondering about those in the install test updates.

File: integration-tests/tests/snap_find_test.go [hunk 3]
```diff
@@ -47,3 +49,69 @@ func (s *searchSuite) TestSearchMustPrintMatch(c *check.C) {
...
+		".*"
+
+	actual := cli.ExecCommand(c, "snap", "find", "--help")
+
+	c.Assert(actual, check.Matches, expected)
+}
+
+// SNAP_FIND_007: find packages with search string containing special characters
+func (s *searchSuite) TestFindWithSpecialCharsInSearchString(c *check.C) {
+	c.Skip("Reenable when LP: 1583952 is fixed")
```
Comment (niemeyer) [inline integration-tests/tests/snap_find_test.go]: Let's please either kill this test or have one that enforces the behavior we want. It doesn't make much sense to have a test (skipped or not) that reinforces bad behavior.

**Comment by niemeyer** [comment]
LGTM assuming the test reinforcing bad behavior is removed.

File: integration-tests/tests/snap_find_test.go (hunk 3 above)
Comment (fgimenez) [inline integration-tests/tests/snap_find_test.go]: ok, removed, thx!

**Comment by elopio** [comment]
Thank  you Federico! Merging...

**Pull request merged**